File: packages/shared/src/types/Song.ts
export interface SongMetadata {
  title: string;
  artist: string;
  album?: string;
  duration: number; // in seconds
  year?: number;
  genre?: string;
  imageUrl?: string; // Album artwork
}

export interface SpotifyTrack {
  id: string; // Spotify track ID
  uri: string; // Spotify URI (spotify:track:xxx)
  name: string;
  artists: Array<{ id: string; name: string }>;
  album: {
    id: string;
    name: string;
    images: Array<{ url: string; height: number; width: number }>;
  };
  duration_ms: number;
  preview_url?: string;
}

export interface Song {
  id: string;
  spotifyTrackId: string; // Spotify track ID
  spotifyUri: string; // Spotify URI for playback
  metadata: SongMetadata;
  answer: {
    title: string;
    artist: string;
  };
  previewUrl?: string; // 30-second preview URL (fallback)
}

File: packages/quiz_master_flutter/lib/models/song.dart
class Song {
  final String id;
  final String title;
  final String artist;
  final String album;
  final String spotifyUri;
  final String? previewUrl;
  final int durationMs;

  Song({
    required this.id,
    required this.title,
    required this.artist,
    required this.album,
    required this.spotifyUri,
    this.previewUrl,
    required this.durationMs,
  });

  factory Song.fromJson(Map<String, dynamic> json) {
    return Song(
      id: json['id'] ?? '',
      title: json['title'] ?? json['name'] ?? '',
      artist: json['artist'] ?? '',
      album: json['album'] ?? '',
      spotifyUri: json['spotifyUri'] ?? json['uri'] ?? '',
      previewUrl: json['previewUrl'],
      durationMs: json['durationMs'] ?? json['duration_ms'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'artist': artist,
      'album': album,
      'spotifyUri': spotifyUri,
      'previewUrl': previewUrl,
      'durationMs': durationMs,
    };
  }
}
